docs/HACKING.md

```diff
@@ -291,3 +291,41 @@ the `OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE` environment variable like so:
 ```
 OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE=quay.io/user/origin-release:v{version number} bin/openshift-install create cluster --log-level=debug
 ```
+
+## Hacking on `machine-os-content`
+
+If you own part of the operating system (from kernel to kubelet) you
+are part of the `machine-os-content`.  More information in [OSUpgrades.md].
+You will want a workflow for testing changes to a cluster.
+
+By far the simplest in a lot of cases is going to be to use `oc debug node/`
+to start by testing your changes on e.g. a single worker node.
+Commands to use here are e.g. `rpm-ostree usroverlay` and/or `rpm-ostree override replace`.
+The first one gives you a writable overlayfs on `/usr` and you can go to town
+replacing binaries there (e.g. `/usr/bin/crio`) or whatever.  If you want to install
+a test kernel in particular you'll need to use `rpm-ostree override replace`.
+
+A much more advanced (but "correct") flow is to build a custom `machine-os-content`
+container.  For that, see https://github.com/coreos/coreos-assembler/pull/489
+(A future iteration of this document will better describe this part)
+But let's assume you have a custom container and have pushed to a registry, for
+this example `quay.io/example/machine-os-content:latest`.
+
+The even more "official" way to do this would be to use the above guide
+to build a custom release image which has that content as an override.
+
+However, if you want to shortcut that, you have a few options.  First,
+you can again use `oc debug node/` and use `pivot` to directly switch
+to the target oscontainer, e.g. `pivot quay.io/example/machine-os-content:latest`.
+If you choose this path though the MCD will go degraded until you revert the change.
```

Comment:
Do we really want to suggest this then? Maybe just use the below suggestions?

Reply:
is there a pro that makes this a good choice?

Reply:
Building a release image each time feels cumbersome basically?  I guess for the same reason I use `make deploy-$x` a lot instead of building a custom release image.